### DISCUSSION TITLE
[Feature] Transcoding off by default + clearer transcoding settings + delete all transcoded videos option

### DISCUSSION THREAD

**Discussion by 825i** [immich-app/immich#10993] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

As posted previously here in this closed thread:

https://github.com/immich-app/immich/discussions/4752#discussioncomment-10007966

>Can we please just have an option in the settings SPECIFICALLY that says "Delete all transcoded videos" with an "Are you sure?" confirmation.

>Please please please!

>Hunting through my entire library to try and identify which videos are transcoded or not is a total nightmare. This setting was ON by default when I installed Immich and it transcoded MANY videos without my knowledge...

>I would also put the setting "Transcode Policy" at the VERY TOP of the settings pane for transcoding instead of over halfway down the bottom of the page (where it currently is). Transcoding is a massively resource intensive task which otherwise grinds your server to a near total halt. For those of us with thousands or even tens of thousands of videos, even if we ticked every single option to not transcode eg. x264, HEVC and so on. There can still be thousands of older videos cough quicktime cough that will be transcoded...

>Please make the settings more user-friendly. The fact that this even forced-on by default is also something I argue against.

>Please and thank you!

As Immich becomes more popular this will become more and more of a complaint by people that transcoding for some reason is on by default. Most of us are really happy with whatever codec our videos are in and don't want that changed for just "compatibility". I would argue that 90% of players and services these days are compatible with again 90% of formats. So whilst I am sure we are all extremely grateful for transcoding even being a feature (really thank you!) please let's have it disabled by default and the settings far more clear. Especially the option to delete all transcoded videos in the Administration settings would be amazing.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by zackpollard** [comment]
Your mention of compatibility is just untrue, although I wish it were true. Firefox, one of the big browsers for example does not yet support hevc codec or mkv video container natively. iPhones are very picky with combination of container, audio and video sources. Older devices may not have hardware encoding for video. I still believe enabling transcoding by default gives the best experience out of the box and we will continue to improve on it.

However, if you want to delete your transcoded files it's really as simple as setting the transcode policy to none and then going to the jobs queue and running the transcoding job for all assets, it will apply your new policy to all assets, deleting transcodes where they exist and are no longer needed.

**Reply by IoSonoAndreaZ** [reply]
I see plex transcodes on the fly if/when needed, and the transcoded videos are temporary and not kept. Can this be implemented in Immich?

**Reply by zackpollard** [reply]
> I see plex transcodes on the fly if/when needed, and the transcoded videos are temporary and not kept. Can this be implemented in Immich?

Potentially in future, it's very complex for a number of reasons and not really the focus of our efforts right now.

**Reply by daaain** [reply]
I agree with your reasoning, but I think it would still be good to make the policy be the top setting and maybe worded clearer, I had to search on the internet to find out about it.

**Comment by OrbitingOcelot** [comment]
So I was just about to ask about on-the-fly transcoding feature when I found this discussion 😀

I have TONS of videos that I definitely do not want to pre-transcode. Instead, if we view a video every once in a while, I would really prefer the transcoding to happen on the fly, and the transcode being cleaned up afterwards. Otherwise it will just be a huge waste of resources, and mess up my backups.

Understand it is not a priority right now, but I would really find this extremely useful. Hope it happens some day ☺

**Reply by IoSonoAndreaZ** [reply]
agree, this is a must have for me

**Comment by hugo-id** [comment]
This is a problem for me.

I have over 1gb of transcodes, on a 15gb database, but the option is turned off, and I have tried re-scanning / re-running transcode job.

I think it could be live videos, but why is this such a fight to handle such a fundamental aspect?

Extremely tedious coming from emby/jellyfin.

**Reply by hugo-id** [reply]
To update, I've now got over 10GB of transcodes (with transcodes "disabled" in the settings) on a 100GB library.

I'm now just sudo rm -fr and hoping Immich won't break / start transcoding things again|

This is my setting:
<img width="786" height="206" alt="{FBA8C191-E923-4F51-8D13-FFFA5F76BE52}" src="https://github.com/user-attachments/assets/fb626269-6e73-45f9-a3bb-21ab9884ca37" />
